src/main/kotlin/mate/academy/model/Figures.kt

```diff
@@ -0,0 +1,46 @@
+package mate.academy.model
+
+import kotlin.math.PI
+
+sealed class Figure(protected val color: Color) : Drawable {
+    protected fun getName() = this::class.simpleName!!.replace(Regex("(?<=\\w)([A-Z])"), " $0").lowercase()
```

Comment:
Quite creative approach, however it is better to simplify it with using enums for figure types, similar to Color.
Then you may use just property name on enum's item.

Also note that using unsafe call is bad practice

Reply:
Let me disagree with you. Adding enum with figure type doesn't simplify the code, on the contrary, it leads to a) unnecessary info reduplication b) it allows such an invalid result as class Circle with type RECTANGLE, that needs more additional code to resolve such situations

Reply:
About unsafe call: in this case it is ok, as I think, because we got null calling this method, according to it's description, 'if the class has no name (if, for example, it is a class of an anonymous object)'. So we can be sure, that null won't be returned